pynest/examples/eprop_plasticity/online_eprop_supervised_regression_neuromorphic_mnist.py

```diff
@@ -363,34 +371,64 @@ def calculate_glorot_dist(fan_in, fan_out):
+    for i in range(n_rec):
+        w = weights_rec_rec[j, i]
+        if np.abs(w) > 0:
+            params_syn_rec["weight"] = w
+            nest.Connect(nrns_rec[i], nrns_rec[j], params_conn_one_to_one, params_syn_rec)
+
+# Sparse Connectivity from Recurrent to Output Neurons
 for j in range(n_out):
     for i in range(n_rec):
         w = weights_rec_out[j, i]
         if np.abs(w) > 0:
             params_syn_out["weight"] = w
             nest.Connect(nrns_rec[i], nrns_out[j], params_conn_one_to_one, params_syn_out)
+
 nest.Connect(nrns_out, nrns_rec, params_conn_all_to_all, params_syn_feedback)  # connection 5
 nest.Connect(gen_rate_target, nrns_out, params_conn_one_to_one, params_syn_rate_target)  # connection 6
 nest.Connect(gen_learning_window, nrns_out, params_conn_all_to_all, params_syn_learning_window)  # connection
 nest.Connect(mm_out, nrns_out, params_conn_all_to_all, params_syn_static)
 
+nest.Connect(nrns_in + nrns_rec, sr, params_conn_all_to_all, params_syn_static)
+
+nest.Connect(mm_rec, nrns_rec_record, params_conn_all_to_all, params_syn_static)
+
 # After creating the connections, we can individually initialize the optimizer's
 # dynamic variables for single synapses (here exemplarily for two connections).
 
-# nest.GetConnections(nrns_rec[0], nrns_rec[1:3]).set([params_init_optimizer] * 2)
+nest.GetConnections(nrns_rec[0], nrns_rec[1:3]).set([params_init_optimizer] * 2)
 
 # %% ###########################################################################################################
 # Create input and output
 # ~~~~~~~~~~~~~~~~~~~~~~~
+# This section involves downloading the Neuromorphic-MNIST (N-MNIST) dataset, extracting it, and preparing it for
+# use in neural network training and testing. The dataset consists of two main components: training and test sets.
+
+# The `download_and_extract_dataset` function handles the retrieval of the dataset from a given URL and its
+# subsequent extraction into a specified directory. It checks for the presence of the dataset to avoid
+# re-downloading. After downloading, it extracts the main dataset zip file, followed by further extraction of
+# nested zip files for training and test data, ensuring that the dataset is ready for loading and processing.
+
+# The `load_image` function reads a single image file from the dataset, converting the event-based neuromorphic
+# data into a format suitable for processing by spiking neural networks. It filters events based on specified pixel
+# blacklists, arranging the remaining events into a structured format representing the image.
+
+# The `DataLoader` class facilitates the loading of the dataset for neural network training and testing.
+# It supports selecting specific labels for inclusion, allowing for targeted training on subsets of the dataset.
+# The class also includes functionality for random shuffling and grouping of data, ensuring diverse and
+# representative samples are used throughout the training process.
```

Comment:
```suggestion
# This section involves downloading the Neuromorphic-MNIST (N-MNIST) dataset, extracting it, and preparing it for
# neural network training and testing. The dataset consists of two main components: training and test sets.

# The `download_and_extract_dataset` function handles the retrieval of the dataset from a given URL and its
# subsequent extraction into a specified directory. It checks for the presence of the dataset to avoid
# re-downloading. After downloading, it extracts the main dataset zip file, followed by further extraction of
# nested zip files for training and test data, ensuring that the dataset is ready for loading and processing.

# The `load_image` function reads a single image file from the dataset, converting the event-based neuromorphic
# data into a format suitable for processing by spiking neural networks. It filters events based on specified pixel
# blocklists, arranging the remaining events into a structured format representing the image.

# The `DataLoader` class facilitates the loading of the dataset for neural network training and testing.
# It supports selecting specific labels for inclusion, allowing for targeted training on subsets of the dataset.
# The class also includes functionality for random shuffling and grouping of data, ensuring diverse and
# representative samples are used throughout the training process.
```

Reply:
Thank you for informing me of this detail. Yes, browsing the Internet it seems that "blocklist" is a standard replacement nowadays.